aggmetric: switch to new btree

RaduBerinde opened the pull request:
Note: first commit is #144555

Use `BtreeG[MetricItem]` in the `aggmetric` code.

Informs: #144504
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/144568)
<!-- Reviewable:end -->

arjunmahishi commented:
@RaduBerinde FYI: In the long term, we were planning swap the `btree` with an Unordered cache. Because we're not really using any tree functionality.

RaduBerinde reviewed (commented):
Sure, that sounds like a good idea, consider this a short term fix so we don't depend on an older btree version.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/144568#-:-ONzipbC7MZJYkgOxZ8u:bdb7p28)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @Abhinav1299, @angles-n-daemons, and @arjunmahishi)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
aggmetric: switch to new btree

Use `BtreeG[MetricItem]` in the `aggmetric` code.

Informs: #144504
Release note: None

RaduBerinde commented:
Friendly ping.

arjunmahishi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/144568#-:-OOOIP4X8ZfqktUcWCXH:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi, @Abhinav1299, and @angles-n-daemons)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+
